errors.go

```diff
@@ -34,6 +40,19 @@ func (e errAuthFailed) Is(target error) bool {
 	return ok
 }
 
+func (e errAuthFailed) ErrorCode() string {
+	errMsg := e.Inner.Error()
+
+	// Find the number in the format of ERR_NGROK_(\d+).
+	reg := regexp.MustCompile(`ERR_NGROK_(\d+)`)
```

Comment:
We shouldn't compile the same regexp over and over.

It also seems wrong that we don't have a proper typed way to get just the errcode part here.
I guess the server isn't providing a properly structured error code?

Shouldn't we be changing the server / lib/tunnel implementation to give us the errcode in a properly structured way, and then implementing the agent sdk changes based on that?

Reply:
Ideally, yes, but I'd prefer to avoid making piecemeal adjustments to the current RPC request/response objects that we have to continue supporting for various agent versions. Someday (hopefully soon?) I want to overhaul them into protobuf messages instead of these hand-rolled json objects, and _then_ we'll make sure to send structured errors.